joomla/joomla-cms
# [5.0] Remove pre-4.0.2 legacy update compatibility

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) removes the handling of legacy files of the Joomla Update component used when updating from 3.10 or 4.0.1 or an older 4.x version to a later 4.y version.

The handling of the legacy updater files had been added to version 4.0.2 with PR #35388 .

In detail the "restore_finalisation.php" had been replaced in the sources by "finalisation.php", and the file "restoration.php" which is generated during the update has been replaced by the generated file "update.php".

All this is removed by this PR here.

See also the text in the comment at the top of the removed "restore_finalisation.php" file:

```
 * This file can be safely removed once a new Joomla version is released which no longer provides an
 * update path from Joomla 3.x and older Joomla 4.0 versions. In theory, that would be Joomla 6.0 or
 * later.
```

As the update path to 5.x has to be from a 4.4 version, this is already fulfilled with 5.0 and not 6.0.

### Testing Instructions

Code review of the changes here and the PR #35388 which had added the removed stuff.

Check that updating the latest 4.4-dev nightly build to the update package built by Drone for this PR here still works.

### Actual result BEFORE applying this Pull Request

There is extra code for support of updating from 4.0.1 or older in the 5.0-dev branch.

### Expected result AFTER applying this Pull Request

There is no extra code for support of updating from 4.0.1 or older in the 5.0-dev branch.

### Link to documentations
Please select:
